### PR TITLE
Comprehensions & core object changes

### PR DESCRIPTION
**Pull request by encukou**
- Finish up Comprehensions
- Write the penultimate chapter

### CODE REVIEW

**Comment by hroncok** [comment]
I'm not 100 % sure with [my latest commit](https://github.com/fedora-python/portingguide/pull/21/commits/b9b74a5fb027ca16741d0df7cfcaa6ab44e93a99), but I think "since" indicates a "perfect" tense. In this case I went with past perfect, because everything with Python 2 is in the past and Python 3 is the present. @torsava is a good candidate to say if I'm right or wrong.

File: source/core-obj-misc.rst [hunk 1]
```diff
@@ -1,19 +1,217 @@
...
+
+
 ``__bool__``
 ~~~~~~~~~~~~
 
+* :ref:`Fixer <python-modernize>`: None
+* Prevalence: Common
+
+Python 2 used the ``__nonzero__`` method to convert an object to boolean,
+i.e. to provide an implementation for :func:`bool`.
```
Comment (hroncok) [inline source/core-obj-misc.rst]: The link here doesn't work.

```rst
:class:`bool`
```

works, but removes the parenthesis and reads bad. Is it possible to fix the link but force the parenthesis? (I'm not that good with RST.)

**Review by hroncok — CHANGES_REQUESTED** [review]
The bool thingy. Also, please check my commits, especially the one I've sent comment about.

File: source/core-obj-misc.rst (hunk 1 above)
Comment (encukou) [reply source/core-obj-misc.rst]: ReST (Sphinx, actually) has syntax for completely custom link text, so I went with that:

    :class:`bool() <bool>`

**Comment by encukou** [comment]
As for the tenses, I trust your judgement. I didn't learn English in school, so I mostly don't care about language technicalities. But big thanks for your attention to detail!

**Comment by torsava** [comment]
@hroncok:
> I'm not 100 % sure with my latest commit, but I think "since" indicates a "perfect" tense. In this case I went with past perfect, because everything with Python 2 is in the past and Python 3 is the present. @torsava  is a good candidate to say if I'm right or wrong.

I believe you are right. Good catch!

**Pull request merged**